tests/test_utilities.py

```diff
@@ -882,14 +883,16 @@ def test_get_course_track_selection_url_raises_exception(self):
         with raises(KeyError):
             utils.get_course_track_selection_url({}, {})
 
-    @mock.patch('integrated_channels.integrated_channel.course_metadata.get_course_runs')
-    def test_base_course_exporter_serialized_data_raises(self, mock_get_course_runs):
-        mock_get_course_runs.return_value = []
+    @mock.patch('enterprise.api_client.enterprise.EnterpriseApiClient.get_enterprise_course_runs')
+    def test_course_exporter_export(self, mock_get_enterprise_course_runs):
+        """
+        ``CourseExporter``'s ``export`` produces a JSON dump of the course data.
+        """
+        mock_get_enterprise_course_runs.return_value.values.return_value = []
         mock_user = mock.MagicMock()
-        mock_plugin_configuration = mock.MagicMock()
-        exporter = BaseCourseExporter(mock_user, mock_plugin_configuration)
-        with raises(NotImplementedError):
-            exporter.get_serialized_data()
+        mock_enterprise_configuration = mock.MagicMock()
+        exporter = CourseExporter(mock_user, mock_enterprise_configuration)
+        assert next(exporter.export()) == b'[]'
```

Comment:
@UmanShahzad Not sure, but would it be worth adding a test for a scenario where `get_enterprise_course_runs` returns one or more course runs? Returning zero course runs (the scenario that this test covers) seems like a bit of an edge case.

Reply:
Will address this when we begin working on the tests ticket (there's definitely work that has to be done for both SAPSF/Degreed tests, and not just the latter).

Reply:
It looks like since this is just testing the base ``CourseExporter``, that export function is just:

```python
    def export(self):
        yield json.dumps(self.courses, sort_keys=True).encode('utf-8'), 'POST'  # I added the 'POST' recently and updated the test
```

Since we don't need to test that `json.dumps` works properly, I think this is actually okay. But I have no clue why this is in `tests/test_utilities.py`, so I'm moving it to the right place.

Reply:
Ah and luckily in the new place I'm moving it to, we have fake course data available easily, so I'll now use that data in this test.